src/robot/libraries/String.py

```diff
@@ -587,10 +590,16 @@ def generate_random_string(self, length=8, chars='[LETTERS][NUMBERS]'):
         | ${low} = | Generate Random String | 12 | [LOWER]         |
         | ${bin} = | Generate Random String | 8  | 01              |
         | ${hex} = | Generate Random String | 4  | [NUMBERS]abcdef |
+        | ${rnd} = | Generate Random String | 5-10 | # Generates a string 5 to 10 characters long |
         """
         if length == '':
             length = 8
-        length = self._convert_to_integer(length, 'length')
+        if '-' in str(length):
+            min_length, max_length = length.split('-')
```

Comment:
Error checking needed in case the value contains more than one `-`. Probably try/except around this assignment and raising an exception with a descriptive error message is easiest.

Should also add tests for cases like `foo-bar` and `-10`. Most likely `_convert_to_integer` already handles them fine, but explicit tests make that sure.

We could actually make `-10` same as `0-10`, but since it looks like a negative number, explicitly adding `0` is easy, and range like `10-` doesn't make any sense I doubt that's a good idea.

Reply:
`-10` is actually parsed currently normally by `_convert_to_integer`. In that case, the keyword doesn't do anything as string length cannot be negative.

This behaviour can be persisted by adding a try-except for an `AttributeError` to `length.split('-')`. Although a negative length string doesn't make any sense, but that's the current behaviour anyway.

Reply:
It's actually a bit questionable that negative numbers work, we could make it an error and require 0 when an empty string is ok, but that's not related to this issue.